pkg/kubectl/cmd/edit.go

```diff
@@ -211,11 +221,11 @@ outter:
 				w = crlf.NewCRLFWriter(w)
 			}
 			if err := results.header.writeTo(w); err != nil {
-				return preservedFile(err, results.file, errOut)
+				allErrs = append(allErrs, preservedFile(err, results.file, errOut))
 			}
 			if !containsError {
 				if err := printer.PrintObj(obj, w); err != nil {
-					return preservedFile(err, results.file, errOut)
+					allErrs = append(allErrs, preservedFile(err, results.file, errOut))
```

Comment:
I believe this should be aggregated into `allErrs` because if a `PrintObj` fails during the editing of the file, then any remaining files in the directory will not be processed.